File: java/rmtest-core/src/main/java/se/redmind/rmtest/selenium/grid/TestBase.java
package se.redmind.rmtest.selenium.grid;

import java.util.Arrays;
import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

import org.junit.runners.Parameterized;
import org.openqa.selenium.WebDriver;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import se.redmind.rmtest.DriverWrapper;

public abstract class TestBase {

    protected final Logger logger = LoggerFactory.getLogger(this.getClass());
    protected final DriverWrapper<?> driverWrapper;
    protected WebDriver webDriver;

    public TestBase(DriverWrapper<?> driverWrapper, final String driverDescription) {
        this.driverWrapper = driverWrapper;
    }

    private static List<Object> getDrivers() {
        return Arrays.asList(DriverProvider.getDrivers());
    }

    @Parameterized.Parameters(name = "{1}")
    public static Collection<Object[]> drivers() {
        return getDrivers().stream().map(obj -> new Object[]{obj, obj.toString()}).collect(Collectors.toList());
    }

}

File: dotNet/RMTest/RMTest/TestBase.cs
using System;
using System.Collections.Generic;
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Remote;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RMTest
{
    public abstract class TestBase
    {
        //protected ICollection<Object[]> allDrivers;
        protected DriverNamingWrapper driverNamingWrapper;
	    protected IWebDriver webDriver;

        public static void main()
        {

        }

        private static List<Object> getDrivers()
        {
            return new List<Object>(DriverProvider.getDrivers());
        }

        //@Parameterized.Parameters(name = "{1}")
	    public static ICollection<Object[]> drivers()
        {
            List<Object[]> driverList = new List<object[]>();
            
            //return getDrivers().stream().map(obj-> new Object[] { obj, obj.toString() }).collect(Collectors.toList());
            foreach (var driver in getDrivers())
            {
                driverList.Add(new object[] { driver, driver.ToString() });
            }
            
            return driverList;
        }

        public static IEnumerable<TestCaseData> TestData()
        {
            {
                DriverNamingWrapper dnw;
                foreach (var driver in drivers())
                {
                    dnw = (DriverNamingWrapper)driver[0]; 
                    yield return new TestCaseData(dnw, driver[1])
                        .SetName(dnw.getCapabilities().BrowserName.ToUpper() + ": (" + dnw.getDescription() + ")")
                        .SetCategory("TestCases");
                }
                
            }
        }
        public TestBase()
        {
            
            //this.allDrivers = drivers();
        }

        //public TestBase(DriverNamingWrapper driverWrapper, String driverDescription)
        //{
        //   this.driverNamingWrapper = driverWrapper;
            // this.deviceType = deviceType;
            // this.initi|alUrl = initialUrl;
        //}
    }
}
